src/vos/vos_pool_scrub.c

```diff
@@ -359,6 +358,16 @@ sc_handle_corruption(struct scrub_ctx *ctx)
 {
 	int rc;
 
+	/** It's ok if we do the checksum calculation after a yield, hoping for the best, but we
+	 *  absolutely must check before modifying data at the current iterator position.  If the
+	 *  entry has been deleted, we can ignore any corruption we found and move on.
+	 */
```

Comment:
Indeed, if an iter callback yield then continue to access the tree namespace, seems it has to explicitly call some validate API to see if the tree or parent tree has been deleted.